server/emails/templates/BaseEmail.tsx

```diff
@@ -101,11 +104,43 @@ export default abstract class BaseEmail<
       return;
     }
 
+    let references: string[] | undefined;
+
+    if (notification) {
+      const prevNotifications = await Notification.unscoped().findAll({
+        attributes: ["messageId"],
+        where: {
+          documentId: notification.documentId,
```

Comment:
I would suggest potentially an email thread for comments/mentions and a separate thread for document created/updated, we can include that logic inside the new method. If the event does not fall into those categories it can return an empty array of references.